clang/lib/Sema/SemaType.cpp

```diff
@@ -3023,6 +3043,14 @@ static QualType GetDeclSpecTypeForDeclarator(TypeProcessingState &state,
         const unsigned AutoParameterPosition = LSI->TemplateParams.size();
         const bool IsParameterPack = D.hasEllipsis();
 
+        IdentifierInfo *II = D.getIdentifier();
+
+        // Invent param name for constrained parameters
+        IdentifierInfo *Ident = nullptr;
+        if (II && Auto->isConstrained() && isa<TemplateTypeParmType>(T))
+          Ident = SemaRef.InventIdentifier(
+              II->getName(), cast<TemplateTypeParmType>(T)->getIndex());
```

Comment:
You're gonna want to call this even if the parameter has no name (in this case the Name.empty() flow would trigger)